src/cloudant/result.py

```diff
@@ -341,11 +341,13 @@ def __iter__(self):
             raise ResultException(103, invalid_options, self.options)
 
         try:
-            if int(self._page_size) <= 0:
+            self._page_size = int(self._page_size)
+            if self._page_size <= 0:
                 raise ResultException(104, self._page_size)
         except ValueError:
             raise ResultException(104, self._page_size)
 
+        self._page_size = int(self._page_size)
```

Comment:
Didn't you already cast this to an int on L344?

Reply:
Also, you don't need to call `int(self._page_size)` everywhere else right as `self._page_size` is an int now?